### ISSUE TITLE
Failed to run `make compile` on Ubuntu 20.04.

### ISSUE THREAD

**Issue by hongyi-zhao** [abo-abo/swiper#2886]
On Ubuntu 20.04, I try to install Ivy from its git repo according to the guide [here](https://oremacs.com/swiper/#installing-from-the-git-repository), but meet the following error:

```
$ git log -1
commit 040d458bce4a88f37359192061bcea5ebe87007c (HEAD -> master, origin/master, origin/HEAD)
Author: Eder Elorriaga <gexplorer8@gmail.com>
Date:   Mon May 24 11:32:17 2021 +0200

    Prevent overriding completions annotations faces
    
    * ivy.el (ivy--format-minibuffer-line): Append
    ivy-completions-annotations face (PR #2875).

$ make compile 
emacs -batch -l elpa.el -L . -f batch-byte-compile colir.el ivy-faces.el ivy-overlay.el ivy.el ivy-avy.el swiper.el counsel.el

In toplevel form:
ivy-avy.el:31:1: Error: Cannot open load file: No such file or directory, avy
make: *** [Makefile:18: compile] Error 1
```
But the make command with other options can be run as shown below:
```
$ make 
emacs -batch -l elpa.el -l colir.el -l ivy-overlay.el -l ivy.el -l swiper.el -l counsel.el -l ivy-test.el -f ivy-test-run-tests
GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0)
 of 2021-06-11
Running 1 tests (2021-06-14 17:15:23+0800, selector ‘ivy--lazy-load-ffap--ffap-url-p’)
   passed  1/1  ivy--lazy-load-ffap--ffap-url-p (0.008456 sec)

Ran 1 tests, 1 results as expected, 0 unexpected (2021-06-14 17:15:23+0800, 0.008548 sec)

Running 75 tests (2021-06-14 17:15:23+0800, selector ‘(not ivy--lazy-load-ffap--ffap-url-p)’)
   passed   1/75  colir-blend-face-background (0.000079 sec)
   passed   2/75  colir-color-parse (0.000062 sec)
   passed   3/75  counsel--elisp-to-pcre (0.000252 sec)
   passed   4/75  counsel--grep-regex (0.000131 sec)
   passed   5/75  counsel--normalize-grep-match (0.000157 sec)
   passed   6/75  counsel--split-command-args (0.000063 sec)
   passed   7/75  counsel--split-string-with-eol-cr (0.000056 sec)
   passed   8/75  counsel--split-string-with-eol-crlf (0.000056 sec)
   passed   9/75  counsel--split-string-with-eol-lf (0.000047 sec)
   passed  10/75  counsel-find-file-single-match-directories (0.152135 sec)
   passed  11/75  counsel-find-file-with-dollars (0.032555 sec)
   passed  12/75  counsel-find-file-with-dotfiles (0.048809 sec)
   passed  13/75  counsel-find-file-with-spaces (0.082566 sec)
   passed  14/75  counsel-url-expand (0.000091 sec)
   passed  15/75  counsel-yank-pop (0.001983 sec)
   passed  16/75  ivy--break-lines (0.000252 sec)
   passed  17/75  ivy--filter (0.000134 sec)
   passed  18/75  ivy--format (0.000102 sec)
   passed  19/75  ivy--handle-directory (0.000043 sec)
   passed  20/75  ivy--handle-full-path-yank-on-remote (0.016431 sec)
   passed  21/75  ivy--minibuffer-index-bounds (0.000073 sec)
   passed  22/75  ivy--preselect-index (0.000097 sec)
   passed  23/75  ivy--regex (0.000195 sec)
   passed  24/75  ivy--regex-fuzzy (0.000107 sec)
   passed  25/75  ivy--regex-ignore-order (0.000208 sec)
   passed  26/75  ivy--regex-plus (0.000074 sec)
   passed  27/75  ivy--sort-function (0.000188 sec)
   passed  28/75  ivy--split (0.000319 sec)
   passed  29/75  ivy--split-negation (0.000128 sec)
   passed  30/75  ivy--split-spaces (0.000231 sec)
   passed  31/75  ivy--yank-handle-case-fold (0.000044 sec)
   passed  32/75  ivy-avy (0.015893 sec)
   passed  33/75  ivy-backward-kill-word (0.002744 sec)
   passed  34/75  ivy-completing-read (0.012122 sec)
   passed  35/75  ivy-completing-read-def-handling (0.033841 sec)
   passed  36/75  ivy-completing-read-handlers (0.015495 sec)
   passed  37/75  ivy-completion-common-length (0.000053 sec)
   passed  38/75  ivy-completion-in-region (0.013514 sec)
   passed  39/75  ivy-counsel-read-directory-name (0.002650 sec)
   passed  40/75  ivy-empty-directory-open (0.018680 sec)
   passed  41/75  ivy-inhibit-action (0.002009 sec)
   passed  42/75  ivy-multi-resume (0.031288 sec)
   passed  43/75  ivy-no-prefix-arg (0.030231 sec)
   passed  44/75  ivy-numeric-prefix-arg (0.002904 sec)
   passed  45/75  ivy-one-prefix-arg (0.032240 sec)
   passed  46/75  ivy-partial-1 (0.014682 sec)
   passed  47/75  ivy-partial-2 (0.017805 sec)
   passed  48/75  ivy-partial-files (0.002853 sec)
   passed  49/75  ivy-re-match (0.000071 sec)
   passed  50/75  ivy-read (0.072490 sec)
   passed  51/75  ivy-read-alist-multi-cands (0.013500 sec)
   passed  52/75  ivy-read-directory-name (0.035053 sec)
   passed  53/75  ivy-read-file-name-in-buffer-visiting-file (0.002739 sec)
   passed  54/75  ivy-read-file-name-initial-input (0.001609 sec)
   passed  55/75  ivy-read-file-name-make-directory (0.013470 sec)
   passed  56/75  ivy-read-history (0.002177 sec)
   passed  57/75  ivy-read-multi-action-1 (0.001293 sec)
   passed  58/75  ivy-read-multi-action-2 (0.012222 sec)
   passed  59/75  ivy-read-preselect (0.001790 sec)
   passed  60/75  ivy-read-prompt (0.049736 sec)
   passed  61/75  ivy-read-remap (0.001088 sec)
   passed  62/75  ivy-read-sort-alist (0.011758 sec)
   passed  63/75  ivy-read-sort-def (0.000926 sec)
   passed  64/75  ivy-starts-with-dotslash (0.000057 sec)
   passed  65/75  ivy-swiper-wgrep (0.018845 sec)
   passed  66/75  ivy-use-selectable-prompt (0.014279 sec)
   passed  67/75  swiper--isearch-format (0.000514 sec)
   passed  68/75  swiper--re-builder (0.000149 sec)
   passed  69/75  swiper--re-builder-char-fold (0.020137 sec)
   passed  70/75  swiper-isearch (0.023842 sec)
   passed  71/75  swiper-isearch-backward (0.024003 sec)
   passed  72/75  swiper-isearch-backward-backspace (0.013546 sec)
   passed  73/75  swiper-isearch-case-fold (0.017365 sec)
   passed  74/75  swiper-query-replace (0.132855 sec)
   passed  75/75  swiper-thing-at-point (0.003450 sec)

Ran 75 tests, 75 results as expected, 0 unexpected (2021-06-14 17:15:24+0800, 1.079017 sec)

$ make all
emacs -batch -l elpa.el -l colir.el -l ivy-overlay.el -l ivy.el -l swiper.el -l counsel.el -l ivy-test.el -f ivy-test-run-tests
GNU Emacs 28.0.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.20, cairo version 1.16.0)
 of 2021-06-11
Running 1 tests (2021-06-14 17:15:47+0800, selector ‘ivy--lazy-load-ffap--ffap-url-p’)
   passed  1/1  ivy--lazy-load-ffap--ffap-url-p (0.009151 sec)

Ran 1 tests, 1 results as expected, 0 unexpected (2021-06-14 17:15:47+0800, 0.009269 sec)

Running 75 tests (2021-06-14 17:15:47+0800, selector ‘(not ivy--lazy-load-ffap--ffap-url-p)’)
   passed   1/75  colir-blend-face-background (0.000082 sec)
   passed   2/75  colir-color-parse (0.000064 sec)
   passed   3/75  counsel--elisp-to-pcre (0.000260 sec)
   passed   4/75  counsel--grep-regex (0.000137 sec)
   passed   5/75  counsel--normalize-grep-match (0.000165 sec)
   passed   6/75  counsel--split-command-args (0.000066 sec)
   passed   7/75  counsel--split-string-with-eol-cr (0.000058 sec)
   passed   8/75  counsel--split-string-with-eol-crlf (0.000057 sec)
   passed   9/75  counsel--split-string-with-eol-lf (0.000050 sec)
   passed  10/75  counsel-find-file-single-match-directories (0.157655 sec)
   passed  11/75  counsel-find-file-with-dollars (0.032886 sec)
   passed  12/75  counsel-find-file-with-dotfiles (0.047660 sec)
   passed  13/75  counsel-find-file-with-spaces (0.085807 sec)
   passed  14/75  counsel-url-expand (0.000113 sec)
   passed  15/75  counsel-yank-pop (0.002190 sec)
   passed  16/75  ivy--break-lines (0.000291 sec)
   passed  17/75  ivy--filter (0.000158 sec)
   passed  18/75  ivy--format (0.000109 sec)
   passed  19/75  ivy--handle-directory (0.000046 sec)
   passed  20/75  ivy--handle-full-path-yank-on-remote (0.018077 sec)
   passed  21/75  ivy--minibuffer-index-bounds (0.000093 sec)
   passed  22/75  ivy--preselect-index (0.000119 sec)
   passed  23/75  ivy--regex (0.000216 sec)
   passed  24/75  ivy--regex-fuzzy (0.000113 sec)
   passed  25/75  ivy--regex-ignore-order (0.000224 sec)
   passed  26/75  ivy--regex-plus (0.000095 sec)
   passed  27/75  ivy--sort-function (0.000240 sec)
   passed  28/75  ivy--split (0.000352 sec)
   passed  29/75  ivy--split-negation (0.000140 sec)
   passed  30/75  ivy--split-spaces (0.000194 sec)
   passed  31/75  ivy--yank-handle-case-fold (0.000047 sec)
   passed  32/75  ivy-avy (0.017982 sec)
   passed  33/75  ivy-backward-kill-word (0.002997 sec)
   passed  34/75  ivy-completing-read (0.013361 sec)
   passed  35/75  ivy-completing-read-def-handling (0.036012 sec)
   passed  36/75  ivy-completing-read-handlers (0.016894 sec)
   passed  37/75  ivy-completion-common-length (0.000071 sec)
   passed  38/75  ivy-completion-in-region (0.015422 sec)
   passed  39/75  ivy-counsel-read-directory-name (0.003070 sec)
   passed  40/75  ivy-empty-directory-open (0.020248 sec)
   passed  41/75  ivy-inhibit-action (0.002503 sec)
   passed  42/75  ivy-multi-resume (0.034632 sec)
   passed  43/75  ivy-no-prefix-arg (0.033291 sec)
   passed  44/75  ivy-numeric-prefix-arg (0.003198 sec)
   passed  45/75  ivy-one-prefix-arg (0.035356 sec)
   passed  46/75  ivy-partial-1 (0.015764 sec)
   passed  47/75  ivy-partial-2 (0.022324 sec)
   passed  48/75  ivy-partial-files (0.003652 sec)
   passed  49/75  ivy-re-match (0.000082 sec)
   passed  50/75  ivy-read (0.070749 sec)
   passed  51/75  ivy-read-alist-multi-cands (0.014685 sec)
   passed  52/75  ivy-read-directory-name (0.037848 sec)
   passed  53/75  ivy-read-file-name-in-buffer-visiting-file (0.002850 sec)
   passed  54/75  ivy-read-file-name-initial-input (0.001815 sec)
   passed  55/75  ivy-read-file-name-make-directory (0.014499 sec)
   passed  56/75  ivy-read-history (0.002533 sec)
   passed  57/75  ivy-read-multi-action-1 (0.001284 sec)
   passed  58/75  ivy-read-multi-action-2 (0.013714 sec)
   passed  59/75  ivy-read-preselect (0.001937 sec)
   passed  60/75  ivy-read-prompt (0.054487 sec)
   passed  61/75  ivy-read-remap (0.001175 sec)
   passed  62/75  ivy-read-sort-alist (0.013026 sec)
   passed  63/75  ivy-read-sort-def (0.001050 sec)
   passed  64/75  ivy-starts-with-dotslash (0.000077 sec)
   passed  65/75  ivy-swiper-wgrep (0.020035 sec)
   passed  66/75  ivy-use-selectable-prompt (0.015718 sec)
   passed  67/75  swiper--isearch-format (0.000657 sec)
   passed  68/75  swiper--re-builder (0.000247 sec)
   passed  69/75  swiper--re-builder-char-fold (0.021817 sec)
   passed  70/75  swiper-isearch (0.026279 sec)
   passed  71/75  swiper-isearch-backward (0.026679 sec)
   passed  72/75  swiper-isearch-backward-backspace (0.014877 sec)
   passed  73/75  swiper-isearch-case-fold (0.018728 sec)
   passed  74/75  swiper-query-replace (0.145189 sec)
   passed  75/75  swiper-thing-at-point (0.003564 sec)

Ran 75 tests, 75 results as expected, 0 unexpected (2021-06-14 17:15:48+0800, 1.153950 sec)
```
Any hints for this problem?

Regards,
HY

**Comment by basil-conto** [comment]
Thanks, is this the same issue as #2583?  TL;DR: It should work if you run `make deps` first.

**Comment by hongyi-zhao** [comment]
> Thanks, is this the same issue as #2583? 

Yes.

> TL;DR: It should work if you run make deps first.

Yes. 

Then, what's the differences between these different make methods?

**Comment by basil-conto** [comment]
>> is this the same issue as #2583?

> Yes.

Thanks for confirming; closing this one in preference for that one.

> Then, what's the differences among these different make methods?

`make deps` installs the packages `avy`, `hydra`, and `wgrep`, which `ivy` supports and integrates with.

`make compile` byte-compiles `ivy`, `swiper`, `counsel` and related packages.

**Comment by hongyi-zhao** [comment]
Thank you for clarification on them. In fact, they all declared in [Makefile](https://github.com/abo-abo/swiper/blob/master/Makefile) and according to the code [here](https://github.com/abo-abo/swiper/blob/040d458bce4a88f37359192061bcea5ebe87007c/Makefile#L6), the following two are equivalent:

```
$ make all
$ make test
```
But I don't think this is a good alias. At first glance, `make all` should be used to do all jobs defined in the Makefile, but here it's only used as an alias for test.

**Comment by basil-conto** [comment]
> But I don't think this is a good alias.

Agreed (I'm not its author), but how much does it matter in practice?

> At first glance, `make all` should be used to do all jobs defined in the Makefile

Yes, but not exactly - according to the GNU Coding Standards, `all` should build all aspects of the program; see [`(info "(standards) Standard Targets")`](https://www.gnu.org/prep/standards/html_node/Standard-Targets.html).

**Comment by hongyi-zhao** [comment]
> but how much does it matter in practice?

It just gives me that illusion, so I pointed out.

> see (info "(standards) Standard Targets").

Thank you for letting me know this.
